### PR TITLE
Handle bundleLookupTag in e2e tests

### LINKED ISSUE #303: Handle bundleLookupTag in e2e tests
This issue is the second half of the problem mentioned in #259 
The `bundleLookupTag` is currently hardcoded in the e2e test -
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/562ad731caf5a982c6a96811e9f759f464a55512/test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-with-kcp.yaml#L31

While this works for now, this value should be configurable. A possible approach could be to treat it as a variable and take input from the user - like we do for `CLUSTER_NAME`
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/562ad731caf5a982c6a96811e9f759f464a55512/test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-with-kcp.yaml#L19

**Note:** We truly cannot test this until #231 is fixed. In the meantime, is good to get some configuration flexibility in place.

### PR DESCRIPTION
**Pull request by huchen2021**
This issue is the second half of the problem mentioned in #259.  The bundleLookupTag is currently hardcoded in the e2e test. While this works for now, this value should be configurable. 

Signed-off-by: Hui Chen <huchen@huchen-a01.vmware.com>

Fixes #303

### CODE REVIEW

**Comment by anusha94** [comment]
@huchen2021 Better to link the PR with the `Fixes` keyword. This will help automatically close the corresponding issue.

Fixes #303

**Comment by huchen2021** [comment]
> @huchen2021 Better to link in PR description with the `Fixes` keyword. This will help automatically close the corresponding issue.

OK

File: test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-with-kcp.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ metadata:
   name: ${CLUSTER_NAME}
 spec:
   bundleLookupBaseRegistry: projects.registry.vmware.com/cluster_api_provider_bringyourownhost
-  bundleLookupTag: v0.1.0_alpha.2
+  bundleLookupTag: ${BUNDLE_LOOKUP_TAG}
```
Comment (anusha94) [inline test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-with-kcp.yaml:31]: We might have to update docs as well in these places -
- https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/docs/getting_started.md#create-workload-cluster
- https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/docs/local_dev.md#create-docker-hosts

**Review by anusha94 — APPROVED** [review]
Will open another issue to update the docs with `BUNDLE_LOOKUP_TAG` env variable after next release.

File: test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-with-kcp.yaml (hunk 1 above)
Comment (huchen2021) [reply test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-with-kcp.yaml:31]: Already update [create-docker-hosts.md](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/docs/local_dev.md#create-docker-hosts). As for [create-workload-cluster.md](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/docs/getting_started.md#create-workload-cluster), it not affected by test/e2e/data/infrastructure-provider-byoh/v1beta1/cluster-template.yaml, it affected by https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/download/<our_new_release>/cluster-template.yaml, Let change it after the release.

**Review by dharmjit — APPROVED** [review]
LGTM

**Pull request merged**